trikot-viewmodels-declarative/viewmodels-declarative/src/commonMain/kotlin/com/mirego/trikot/viewmodels/declarative/components/impl/VMDButtonViewModelImpl.kt

```diff
@@ -17,19 +19,19 @@ open class VMDButtonViewModelImpl<C : VMDContent>(
     defaultContent: C
 ) : VMDButtonViewModel<C>(cancellableManager) {
 
-    override var actionBlock: () -> Unit = {}
+    private val atomicReferenceActionBlock = AtomicReference { }
+
+    override val actionBlock: () -> Unit get() = atomicReferenceActionBlock.value
```

Comment:
```suggestion
    override var actionBlock: () -> Unit by atomic { }
```

You could use the `by atomic()` extension to keep using a var without the need of a separate variable